### PR TITLE
Adds vault IDs to Vault credential chips

### PR DESCRIPTION
**Pull request by AlexSCorey**
##### SUMMARY
This addresses #8532 This adds the vault ID to the vault credential name.  I show it the same way that it's shown in the old ui.  However, since we display the id each time we display the credential name I wonder if we should just build the name with the ID when the credential is created and save it that way. That would potentially save several api calls we need to make the get the vault id and it would eliminate places where we build the name in the ui....on `MultiCredentialsLookup.jsx` and `CredentialChip.jsx`.

There are a few places where this hasn't been implemented because those parts are actively in flight right now. https://github.com/ansible/awx/issues/8735

##### ISSUE TYPE
 - Feature Pull Request

##### COMPONENT NAME
 - UI

##### ADDITIONAL INFORMATION
Screen Shot 2020-12-03 at 10 29 11 AM

### CODE REVIEW

File: awx/ui_next/src/components/CredentialChip/CredentialChip.jsx [hunk 1]
```diff
@@ -19,7 +19,9 @@ function CredentialChip({ credential, i18n, i18nHash, ...props }) {
   return (
     <Chip {...props}>
       <strong>{type}: </strong>
-      {credential.name}
+      {credential.kind === 'vault'
+        ? `${credential.name} | ${credential.inputs?.vault_id}`
+        : credential.name}
```
Comment (jakemcdermott) [inline awx/ui_next/src/components/CredentialChip/CredentialChip.jsx]: @AlexSCorey It's possible for a vault credential to have no vault_id, in which case we'd want to hide the `|`

**Review by jakemcdermott — CHANGES_REQUESTED** [review]

File: awx/ui_next/src/screens/Job/Job.jsx [hunk 2]
```diff
@@ -26,20 +26,34 @@ function Job({ i18n, lookup, setBreadcrumb }) {
...
-  const { isLoading, error, request: fetchJob, result } = useRequest(
+  const {
+    isLoading,
+    error,
+    request: fetchJob,
+    result: { response },
+  } = useRequest(
     useCallback(async () => {
       const { data } = await JobsAPI.readDetail(id, type);
+      console.log(data.summary_fields, 'data summary fields');
```
Comment (jakemcdermott) [inline awx/ui_next/src/screens/Job/Job.jsx]:
```suggestion
```

File: awx/ui_next/src/screens/Job/Job.jsx [hunk 3]
```diff
@@ -26,20 +26,34 @@ function Job({ i18n, lookup, setBreadcrumb }) {
...
+        data.summary_fields.credentials = results;
+      }
       setBreadcrumb(data);
-      return data;
+      return { response: data };
     }, [id, type, setBreadcrumb]),
-    null
+    { response: {} }
   );
+  console.log(error, 'error');
```
Comment (jakemcdermott) [inline awx/ui_next/src/screens/Job/Job.jsx]:
```suggestion
```

File: awx/ui_next/src/screens/Project/Project.jsx [hunk 4]
```diff
@@ -44,6 +44,19 @@ function Project({ i18n, setBreadcrumb }) {
...
+        const params = {
+          page: 1,
+          page_size: 200,
+          order_by: 'name',
+        };
+        const {
+          data: { results },
+        } = await ProjectsAPI.readCredentials(data.id, params);
+
+        data.summary_fields.credentials = results;
```
Comment (jakemcdermott) [inline awx/ui_next/src/screens/Project/Project.jsx:58]: @AlexSCorey 
Hypothetically speaking, would we need to make this extra api call if `vault_id` were included in the project summary credentials?

File: awx/ui_next/src/components/Lookup/MultiCredentialsLookup.jsx [hunk 5]
```diff
@@ -71,6 +71,13 @@ function MultiCredentialsLookup(props) {
         loadCredentials(params, selectedType.id),
         CredentialsAPI.readOptions(),
       ]);
+
+      results.map(result => {
+        if (result.kind === 'vault')
+          result.label = `${result.name} | ${result.inputs.vault_id}`;
```
Comment (jakemcdermott) [inline awx/ui_next/src/components/Lookup/MultiCredentialsLookup.jsx:77]: ditto https://github.com/ansible/awx/pull/8738/files#r535449988 (and anywhere else we build the name)

File: awx/ui_next/src/screens/Template/Template.jsx [hunk 6]
```diff
@@ -44,6 +44,19 @@ function Template({ i18n, me, setBreadcrumb }) {
...
+      if (data.summary_fields.credentials) {
+        const params = {
+          page: 1,
+          page_size: 200,
+          order_by: 'name',
+        };
+        const {
+          data: { results },
+        } = await JobTemplatesAPI.readCredentials(data.id, params);
+        console.log(results, 'data');
```
Comment (jakemcdermott) [inline awx/ui_next/src/screens/Template/Template.jsx]:
```suggestion

```

File: awx/ui_next/src/screens/Project/Project.jsx (hunk 4 above)
Comment (AlexSCorey) [reply awx/ui_next/src/screens/Project/Project.jsx:58]: @jakemcdermott 
no.  if we could add the vault_id to the summary_fields then we wouldn't need to make the extra api calls. In conversations with @wenottingham and @ryanpetrello it seems like there are reasons that adding that info is not possible or not advisable.

**Review by jakemcdermott — APPROVED** [review]

**Comment by tiagodread** [comment]
@AlexSCorey found so far:

- [x] - `TypeError: Cannot read property 'status' of null` trying to launch any template 

image

It's possible to set multiple vault credentials with the same vault ID doing:

vault

**Review by tiagodread — APPROVED** [review]
Test covered by https://github.com/ansible/tower-qa/pull/5837
all tests are green in this branch

**Pull request merged**